Update docs about inventory page to show new host details flyout

dedemorton opened the pull request:
Closes https://github.com/elastic/observability-docs/issues/3249, https://github.com/elastic/observability-docs/issues/3250, and https://github.com/elastic/observability-docs/issues/3210

Note that I've rearranged some of the content to improve the flow a bit and avoid confusing users:
* Rearranged the sections into a more logical order. For example, moved the filtering section to the top because it seemed like the first thing users will want to do. Also rearranged the expandable sections for each tab so that they appear in the same order as the UI.
* Added a section title for the content about viewing host metrics to make it easier for users to scan the page.
* Changed some section titles because they weren't specific enough or were misleading.

This restructuring might make the diff kind of confusing, so you'll probably want to read through the previews here:

* https://observability-docs_3328.docs-preview.app.elstc.co/guide/en/observability/master/view-infrastructure-metrics.html 
* https://observability-docs_3328.docs-preview.app.elstc.co/guide/en/observability/master/analyze-hosts.html (updates added Nov 2)

Changes on this page are minor and are best reviewed in the diff:

https://observability-docs_3328.docs-preview.app.elstc.co/guide/en/observability/master/host-metrics.html

TODO:
- [x] Update topic about Inventory page to reflect UI changes (new host details and a few other bits)
- [x] Make some structural tweaks to improve flow.
- [x] Update host metrics page.
- [x] Resolve outstanding questions:
  - [x]  Why does clicking a host in the waffle map vs the table behave differently? In the table view, a popup with links appears (instead of the flyout opening). Is this intentional or a bug?
  - [x] Looks like new pinning capability was added to the Metadata tab. How does this work? Should we mention it in the docs?
  - [x] The section about adding custom metrics isn't working as documented. I don't see the custom metric shown in a chart or in the "group by" menu. Does this feature still work as documented? Am I doing something wrong?
- [x] Update remaining screenshots.
- [x] Update the Hosts page as required.

roshan-elastic reviewed (changes requested):
Hey @dedemorton, thanks for this - looking great!

I can see that the OSQuery and Anomaly tabs use the old screenshots but I imagine you did this because you weren't able to get screenshots in the environment you have?

Could I supply new screenshots for you to use?

> **Note** : I'm supplying with/without headers in case including the headers makes it hard for you to maintain

**Overview**

_with header_
<img width="769" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/08ce15ec-dab7-4dab-bb7c-315a2ae0ed5b">

_without header_
<img width="750" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/81cea894-56b6-4b27-957e-cc77191e1295">

**Metadata**

_with header_
<img width="773" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/5d75b6ca-c6ac-4d51-a94a-2b9120850091">

_without header_
<img width="762" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/4a9f70b0-18b6-49b9-adaa-bd4154ec69af">

**Logs**

_with header_
<img width="770" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/547c0fcf-82af-481e-a0ab-5eca5b663ec9">

_without header_
<img width="756" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/fc9f91e2-7576-4f3e-9fb1-48a8c6ce0ed3">


**Anomalies**

_with header_
![image](https://github.com/elastic/observability-docs/assets/117740680/cf83f8b4-cd23-4e14-9937-4023761d6b53)

_without header_
![image](https://github.com/elastic/observability-docs/assets/117740680/2d430058-ee40-4c18-b6de-4e5060fd37f4)

**OSQuery**

_with header_
<img width="763" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/8ea9597e-2266-4154-8a66-2c958ef2577e">

_without header_
<img width="754" alt="image" src="https://github.com/elastic/observability-docs/assets/117740680/5d1ef32f-193b-4c0b-b787-76cbc9b785ed">

dedemorton commented:
@roshan-elastic I've added the images (thanks again for creating those!) and updated the docs about the hosts page. I decided to include the tab descriptions by importing a shared file.

As we move towards GAing the Hosts page, I should take some time to do more cleanup on this page because I feel we might be describing some details that are obvious to users (like the filtering section). But for now, I wanted to focus on getting the existing content updated for 8.11. Let me know what you think.

roshan-elastic commented:
@dedemorton - as promised:

> Why does clicking a host in the waffle map vs the table behave differently? In the table view, a popup with links appears (instead of the flyout opening). Is this intentional or a bug?

This could be construed as either but either way, it's not a great experience. At the moment, very few people use the table view (and we're actually aiming to completely rebuild inventory) so we're try to avoid investing in the current version of inventory where we can avoid it.

>  Looks like new pinning capability was added to the Metadata tab. How does this work? Should we mention it in the docs?

Yeah, this is pretty cool. In short, when you 'pin' something it will stay at the top in the host view across any hosts (it is saved at the browser-level though so won't persist if you log in from another browser/laptop etc).

Feel free to mention but I feel like it's probably intuitive enough for users to see/understand so might not be worth the effort of maintaining that content.

>  The section about adding custom metrics isn't working as documented. I don't see the custom metric shown in a chart or in the "group by" menu. Does this feature still work as documented? Am I doing something wrong?

For the 'group by', I think this is might be documented incorrectly. When you add a custom metric, the 'group by' will respect the metric you have chosen (inc if it is custom) but the actual group by workflow doesn't involve selecting metrics:

![image](https://github.com/elastic/observability-docs/assets/117740680/30968bea-25b4-46ce-b465-3be762ff3c29)

On the seeing the metric in the chart, did you mean this functionality?

![image](https://github.com/elastic/observability-docs/assets/117740680/f2931f2a-3558-4d85-94f0-466ea347ece1)

If so, we completely missed this functionality. This isn't available in the newer fly-out. We'll prioritise it though if we find users really need it...for now it might be best to remove the docs where it says:

"A chart is also displayed for each [custom metric](https://www.elastic.co/guide/en/observability/current/view-infrastructure-metrics.html#custom-metrics) that you have added and defined on the Inventory page."

Hopefully that's everything?

roshan-elastic reviewed (approved):
Looks good to me!

dedemorton commented:
@roshan-elastic Thanks! I think you are right about the pinning feature, especially since it's a convenience thing...not really necessary for everyone to use.

If you decide to add the custom metric chart to the new flyout, just open a docs issue so we know to update the docs.

mdbirnstiehl commented on docs/en/observability/monitor-infra/host-details-partial.asciidoc:
```diff
@@ -0,0 +1,149 @@
+image::images/anomalies-overlay.png[Anomalies]
+
+The *Anomalies* table displays a list of each single metric {anomaly-detect} job for the specific host. By default, anomaly
+jobs are sorted by time to show the most recent job. 
```
```suggestion
jobs are sorted by time, showing the most recent jobs first. 
```
Maybe not necessary, but I got hung up on this sentence, thinking of "time to show the most recent job" was a way to sort.

mdbirnstiehl commented on docs/en/observability/monitor-infra/host-details-partial.asciidoc:
```diff
@@ -0,0 +1,149 @@
+
+In the example above, we query for the top 5 memory hogs running on the host.
+Under the *Results* tab, the total virtual memory size (`total_size` renamed to
+`memory_used` to be a little more user friendly) is returned in descending order,
```
It looks like the new screenshot doesn't show the **Results** or **Status** tab, so the user won't see it in the example. Not sure if this would maybe be more meaningful showing those tabs, if the tabs still exist?

dedemorton replied:
@mdbirnstiehl Good catch! I should have checked the new screen to make sure it had the query details. Unfortunately this screen requires a setup I don't currently have. When I enter the query shown in the screen capture and click Submit, it doesn't do anything (probably because I haven't installed the agent).

I also think it's not great that we refer to an example query that _only_ appears in the screen capture (not great for accessibility).

To get this merged today, I am going to remove the content that I can't validate and follow up with Roshan tomorrow about getting an updated screen shot. Thanks!

dedemorton replied:
I'm going to leave the bit about the tabs because I'm pretty sure they still exist, but I will confirm when I see the updated screenshot.

Opened an issue to make sure we get that screenshot updated: https://github.com/elastic/observability-docs/issues/3350

mdbirnstiehl reviewed (dismissed):
Just a couple of small suggestions, otherwise LGTM. I'm going to approve even though it will be dismissed if you decide to make any changes 😁
